File: app/src/main/java/com/bkjcb/rqapplication/base/retrofit/DataService.java
package com.bkjcb.rqapplication.base.retrofit;

import com.bkjcb.rqapplication.base.model.BottleResult;
import com.bkjcb.rqapplication.base.model.HttpResult;
import com.bkjcb.rqapplication.userRecord.model.UserInfoResult;
import com.bkjcb.rqapplication.base.model.UserResult;

import io.reactivex.Observable;
import retrofit2.Call;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Query;

/**
 * Created by DengShuai on 2019/10/24.
 * Description :
 */
public interface DataService {
    @GET("/rq/push/getUserInfos")
    Observable<UserInfoResult> getUserInfos(@Query("areaCode")String code);
    @POST("/rq/push/getLoginUser")
    @FormUrlEncoded
    Observable<UserResult> getLoginUser(@Field("userName") String userName, @Field("password")String password); @POST("/rq/push/getLoginUser")
    @FormUrlEncoded
    Call<String> getLogin(@Field("userName") String userName, @Field("password")String password);
    @GET("/rq/push/getBottleData")
    Observable<BottleResult> getBottleData(@Query("userGuid")String Guid);
    @POST("/rq/push/saveTourCheck")
    @FormUrlEncoded
    Observable<HttpResult> saveTourCheck(@Field("userGuid") String Guid,@Field("tourPeople") String name,@Field("tourState") String state,@Field("tourRemark") String remark);

    @POST("/rq/push/handleUserNew")
    @FormUrlEncoded
    Observable<HttpResult> changeUserInfo(@Field("userGuid") String Guid,@Field("userName") String name,@Field("userAddress") String address);

    @POST("/rq/push/handleUserNew")
    @FormUrlEncoded
    Observable<HttpResult> changeUserInfo(@Field("areaCode") String Guid,@Field("userName") String name,@Field("userAddress") String address,@Field("userType")int type);
}

File: app/src/main/java/com/bkjcb/rqapplication/retrofit/DataService.kt
package com.bkjcb.rqapplication.retrofit

import com.bkjcb.rqapplication.model.HttpResult
import com.bkjcb.rqapplication.model.UserResult
import com.bkjcb.rqapplication.treatmentdefect.model.BottleResult
import com.bkjcb.rqapplication.treatmentdefect.model.UserInfoResult
import io.reactivex.Observable
import retrofit2.Call
import retrofit2.http.*

/**
 * Created by DengShuai on 2019/10/24.
 * Description :
 */
interface DataService {
    @GET("/rq/push/getUserInfos")
    fun getUserInfos(@Query("areaCode") code: String?): Observable<UserInfoResult>

    @POST("/rq/push/getLoginUser")
    @FormUrlEncoded
    fun getLoginUser(@Field("userName") userName: String?, @Field("password") password: String?): Observable<UserResult>

    @POST("/rq/push/getLoginUser")
    @FormUrlEncoded
    fun getLogin(@Field("userName") userName: String?, @Field("password") password: String?): Call<String>

    @GET("/rq/push/getBottleData")
    fun getBottleData(@Query("userGuid") Guid: String?): Observable<BottleResult>

    @POST("/rq/push/saveTourCheck")
    @FormUrlEncoded
    fun saveTourCheck(@Field("userGuid") Guid: String?, @Field("tourPeople") name: String?, @Field("tourState") state: String?, @Field("tourRemark") remark: String?): Observable<HttpResult>

    @POST("/rq/push/handleUserNew")
    @FormUrlEncoded
    fun changeUserInfo(@Field("userGuid") Guid: String?, @Field("userName") name: String?, @Field("userAddress") address: String?): Observable<HttpResult>

    @POST("/rq/push/handleUserNew")
    @FormUrlEncoded
    fun changeUserInfo(@Field("areaCode") Guid: String?, @Field("userName") name: String?, @Field("userAddress") address: String?, @Field("userType") type: Int): Observable<HttpResult>
}